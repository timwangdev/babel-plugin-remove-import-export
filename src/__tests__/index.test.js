const { transform } = require('@babel/core');

const example = `
import { LinkedList } from 'some-lib';

function foo() {
  return new LinkedList(['bar']);
}

export class Solution {
  add(a, b) {
    return a + b;
  }
}

export default foo;
`;

it('works with no options', () => {
  const { code } = transform(example, { plugins: [require('../index')] });
  expect(code).toMatchSnapshot();
});

it('works with removeImport false options', () => {
  const { code } = transform(example, {
      plugins: [[require('../index'), { removeImport: false }]]
  });
  expect(code).toMatchSnapshot();
});

it('works with removeExport false options', () => {
  const { code } = transform(example, {
      plugins: [[require('../index'), { removeExport: false }]]
  });
  expect(code).toMatchSnapshot();
});

it('works with removeExportDefault false options', () => {
  const { code } = transform(example, {
      plugins: [[require('../index'), { removeExportDefault: false }]]
  });
  expect(code).toMatchSnapshot();
});


it('works with preseveNamedDeclaration false options', () => {
  const { code } = transform(example, {
      plugins: [[require('../index'), { preseveNamedDeclaration: false }]]
  });
  expect(code).toMatchSnapshot();
});