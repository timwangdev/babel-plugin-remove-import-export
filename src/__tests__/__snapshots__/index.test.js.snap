// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with no options 1`] = `
"function foo() {
  return new LinkedList(['bar']);
}

class Solution {
  add(a, b) {
    return a + b;
  }

}"
`;

exports[`works with preseveNamedDeclaration false options 1`] = `
"function foo() {
  return new LinkedList(['bar']);
}"
`;

exports[`works with removeExport false options 1`] = `
"function foo() {
  return new LinkedList(['bar']);
}

export class Solution {
  add(a, b) {
    return a + b;
  }

}
export default foo;"
`;

exports[`works with removeExportDefault false options 1`] = `
"function foo() {
  return new LinkedList(['bar']);
}

class Solution {
  add(a, b) {
    return a + b;
  }

}

export default foo;"
`;

exports[`works with removeImport false options 1`] = `
"import { LinkedList } from 'some-lib';

function foo() {
  return new LinkedList(['bar']);
}

class Solution {
  add(a, b) {
    return a + b;
  }

}"
`;
